### PR TITLE
docs: make the shell completion page more consistent

### PR DESCRIPTION
**Pull request by rpkoller**
## The Issue
Initial draft to make the shell completion sections more consistent among each other as well as more inline with https://docs.brew.sh/Shell-Completion . Not finished yet. Git Bash, PowerShell and the Tar archive section still need work. But feedback about bash and fish is welcome (as well as how consistent it is with the previous zsh changes)

### CODE REVIEW

**Comment by rfay** [comment]
Rebased

**Comment by rfay** [comment]
Rebased

**Comment by rfay** [comment]
Rebased

File: docs/content/users/install/shell-completion.md [hunk 1]
```diff
@@ -2,22 +2,65 @@
...
+To make Homebrew completions available in Bash the Homebrew-managed path, either `etc/profile.d` or `etc/bash_completion.de` has to be sourced.
 
-```
-[[ -r "$(brew --prefix)/etc/profile.d/bash_completion.sh" ]] && . "$(brew --prefix)/etc/profile.d/bash_completion.sh"
+=== "macOS Bash"
+
+     Add the following block to your `~/.bashprofile` or if it doesn't exist to your `~/.profile` file (see [docs](https://docs.brew.sh/Shell-Completion#configuring-completions-in-bash)):
+
+    ```bash
+    if type brew &>/dev/null
```
Comment (rfay) [inline docs/content/users/install/shell-completion.md:16]: Where did this come from and why is it required? Why the for loop? 

Why would /etc/profile.d not have the bash_completion.sh?

File: docs/content/users/install/shell-completion.md [hunk 2]
```diff
@@ -2,22 +2,65 @@
...
+Shells like Bash and Zsh need help to do this though, they have to know what the options are, and DDEV provides the necessary hint scripts.
 
 ## macOS Bash with Homebrew
 
-The easiest way to use Bash completion on macOS is install it with Homebrew ([docs](https://docs.brew.sh/Shell-Completion#configuring-completions-in-bash)). `brew install bash-completion`. When you install it though, it will warn you with something like this, which **may vary on your system**. Add the following line to your `~/.bash_profile` file (or if that doesn't exist, to your `~/.profile`:
+To make Homebrew completions available in Bash the Homebrew-managed path, either `etc/profile.d` or `etc/bash_completion.de` has to be sourced.
 
-```
-[[ -r "$(brew --prefix)/etc/profile.d/bash_completion.sh" ]] && . "$(brew --prefix)/etc/profile.d/bash_completion.sh"
+=== "macOS Bash"
```
Comment (rfay) [inline docs/content/users/install/shell-completion.md:11]: Why do we have a "macOS Bash" and a "macOS Bash with Bash Completion"? 

image

**Review by rfay — CHANGES_REQUESTED** [review]
I'm afraid this is not an improvement over what we currently have. Maybe it's trying to do too many things?

If this is just about consistency, a new PR could address that. I wasn't able to sort out what this was actually trying to do. 

What we want is for people to be able to quickly and easily get completion working. On macOS that means using homebrew for bash completion.

**Comment by rfay** [comment]
Closing this for now. Thanks for all your work on all these things!

**Comment by rpkoller** [comment]
> I'm afraid this is not an improvement over what we currently have. Maybe it's trying to do too many things?
> 
> If this is just about consistency, a new PR could address that. I wasn't able to sort out what this was actually trying to do.
> 
> What we want is for people to be able to quickly and easily get completion working. On macOS that means using homebrew for bash completion.

not about consistency at all. i was trying to improve the usability of that page. not requiring stuff like bash completion or oh my zsh. save the user the time to read through homebrew instructions and other sources. and that is what is accomplished with the changes. just havent had the time to return and finish the missing  manual step at the end.

File: docs/content/users/install/shell-completion.md (hunk 1 above)
Comment (rpkoller) [reply docs/content/users/install/shell-completion.md:16]: that is from the homebrew docs. that way you dont need the bash completion as stated in the next tab.

File: docs/content/users/install/shell-completion.md (hunk 2 above)
Comment (rpkoller) [reply docs/content/users/install/shell-completion.md:11]: for people that dont want to install bash completion for now (like me back then) you have the installation step taken from homebrew and for those who want more those can install bash completion. more or less the same case like for zsh and oh my zsh.

**Pull request closed without merging**